src/main/scala/org/hammerlab/guacamole/loci/partitioning/CappedRegionsPartitioner.scala

```diff
@@ -131,4 +73,69 @@ class CappedRegionsPartitioner[R <: ReferenceRegion: ClassTag](regions: RDD[R],
 
     LociPartitioning(lociSets)
   }
+
+  private def printStats(coverageRDD: CoverageRDD[R], lociBroadcast: Broadcast[LociSet]): Unit = {
+    val (depthRunsRDD, validLoci, invalidLoci) =
+      coverageRDD.validLociCounts(halfWindowSize, lociBroadcast, maxRegionsPerPartition)
+
+    val numDepthRuns = depthRunsRDD.count
+    val numDepthRunsToTake = 1000
+    val depthRuns =
+      if (numDepthRuns <= numDepthRunsToTake)
+        depthRunsRDD.collect()
+      else
+        depthRunsRDD.take(numDepthRunsToTake)
+
+    val avgRunLength =
+      (for {(_, num) <- depthRuns} yield num.toLong * num).sum.toDouble / validLoci
+
+    val depthRunsByContig =
+      depthRuns
+      .groupBy(_._1._1)
+      .mapValues(_.map {
+        case ((_, valid), num) => num -> valid
+      })
+      .toArray
+      .sorted(new KeyOrdering[ContigName, Array[(NumLoci, Boolean)]](ContigName.ordering))
+
+    val overflowMsg =
+      if (numDepthRuns > numDepthRunsToTake)
+        s". First $numDepthRunsToTake runs:"
+      else
+        ":"
+
+    def runsStr(runsIter: Iterator[(NumLoci, Boolean)]): String = {
+      val runs = runsIter.toVector
+      val rs =
+        (for ((num, valid) <- runs) yield {
+          s"$num${if (valid) "↓" else "↑"}"
+        }).mkString(" ")
+      if (runs.length == 1)
+        s"$rs"
+      else {
+        val total = runs.map(_._1.toLong).sum
+        s"${runs.length} runs, $total loci (avg %.1f): $rs".format(total.toDouble / runs.length)
+      }
+    }
+
+    val totalLoci = validLoci + invalidLoci
+
+    val loci = lociBroadcast.value
+
+    progress(
```

Comment:
Any reason to come back to `progress` here vs going more in the direction of #551